### PR TITLE
feat: Spanner Implementation for Vector Store

### PR DESCRIPTION
**Pull request by gauravpurohit06**
Spanner Implementation for Vector Stores, supports: 

1. Search: (Exact Neighbors Search/ KNN, search with maximum marginal relevance) 
2. Initialization: (Existing table, also supports creation of new table) 
3. Add and Delete Vector documents/embeddings.

### CODE REVIEW

File: src/langchain_google_spanner/vector_store.py
Comment (averikitsch) [inline src/langchain_google_spanner/vector_store.py:206]: This doesn't match the fields listed.

File: src/langchain_google_spanner/vector_store.py
Comment (averikitsch) [inline src/langchain_google_spanner/vector_store.py:1094]: Don't use kwargs and add use comment `# type: ignore[override]` to ignore the warning

**Review by averikitsch — CHANGES_REQUESTED** [review]

File: src/langchain_google_spanner/vector_store.py
Comment (gauravpurohit06) [reply src/langchain_google_spanner/vector_store.py:458]: yes we are overwriting it, as we are fetching all columns from backend and ignoring the columns which are passed by user. I am adding another validation check to see if the final list has the id, content and embedding column.

File: src/langchain_google_spanner/vector_store.py
Comment (gauravpurohit06) [reply src/langchain_google_spanner/vector_store.py:206]: It's an internal field, which we would be deriving from the staleness type and value passed by the user.

File: src/langchain_google_spanner/vector_store.py
Comment (gauravpurohit06) [reply src/langchain_google_spanner/vector_store.py:1094]: In my opinion, we need kwards to dervide the value if user is passing a dictionary and we can't take these values explicitly in the method signature because of interface constraints.

Comment (averikitsch) [reply src/langchain_google_spanner/vector_store.py:1146]: The other implementations are ignoring the interface constraint. I don't think we should support env vars or kwargs

File: src/langchain_google_spanner/vector_store.py
Comment (averikitsch) [inline src/langchain_google_spanner/vector_store.py:341]: Does table name need to be quoted so hyphens can be used

File: src/langchain_google_spanner/vector_store.py
Comment (averikitsch) [inline src/langchain_google_spanner/vector_store.py:377]: Should column names also be in quotes?

**Review by averikitsch — COMMENTED** [review]
Will tests also be provided in this PR?

File: src/langchain_google_spanner/vector_store.py
Comment (gauravpurohit06) [reply src/langchain_google_spanner/vector_store.py:377]: it's not required as Spanner doesn't support hyphens.

File: src/langchain_google_spanner/vector_store.py
Comment (gauravpurohit06) [reply src/langchain_google_spanner/vector_store.py:341]: it's not required as Spanner doesn't support hyphens.

File: src/langchain_google_spanner/vector_store.py
Comment (averikitsch) [reply src/langchain_google_spanner/vector_store.py:1146]: It should be safe to override the interface and add the additional required arguments for instance_id, database_id and table_name (see https://github.com/googleapis/langchain-google-cloud-sql-pg-python/blob/main/src/langchain_google_cloud_sql_pg/vectorstore.py#L390). The developer experience is not good when it errors on missing kwards or needing to set env vars.

File: src/langchain_google_spanner/vector_store.py
Comment (averikitsch) [inline src/langchain_google_spanner/vector_store.py:52]: Since we are waiting on a Spanner review, can we also add the user agent in this PR or separately:
```suggestion
from .version import __version__

USER_AGENT = "langchain-google-spanner-python:vectorstore" + __version__
```

Then update the logic for the client to append the user agent
https://github.com/googleapis/langchain-google-datastore-python/blob/3a62380794ad567ff8e9c0d5d7054093fb54cdb4/src/langchain_google_datastore/document_loader.py#L160-L168

**Review by rahul2393 — APPROVED** [review]

**Review by averikitsch — APPROVED** [review]
Please update the error messages using "id-{" before merging.

**Pull request merged**
